Dockerfile.rp-full-dev

```diff
@@ -0,0 +1,23 @@
+FROM registry.access.redhat.com/ubi8/toolbox:8.10
+
+# Needed for installing packages
+USER root 
+
+RUN dnf install https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm -y \
+    && dnf install make golang azure-cli -y \
+    && dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm -y \
+    && dnf install screen openvpn -y \
+    && dnf install jq docker device-mapper-devel skopeo -y \
+    && dnf clean all -y
+
+ARG ARO_RP_BRANCH
+ENV ARO_RP_DIRECTORY=/home/ARO-RP
+RUN git clone https://github.com/Azure/ARO-RP.git ${ARO_RP_DIRECTORY}\
```

Comment:
if we are choosing to support `git clone` instead of using `COPY` of the current local code, we should add a `Dockerfile.rp-full-dev.dockerignore` with a single line that has `*` to build this without any cache...

but I think my preference would be to use `COPY . .` (and a less restrictive `Dockerfile.rp-full-dev.dockerignore`) for a few reasons:

1. It allows us to test work-in-progress changes live without needing a merged PR
2. It eliminates a dependency from the process (`git`), which has downstream dependencies like `github.com` being online
3. It ties the instructions in the dockerfile to the current commit - that is, if we, say, add a new shell utility dependency used to deploy, we can push those changes here at the same time that the code is also changed and tie it to a specific version we know works
4. Due to item 3 above, `COPY` has a much better chance of succeeding than `git clone`, because now this dockerfile has to accommodate _all_ commits through time, whereas a `COPY` only has to support the _current_ commit or working directory to function (read: no mismatch between code and dependencies or versions of dependencies)

Reply:
Thank you for the thorough explanation. I had doubts about that (mainly reasons 1and 2), and I agree with your suggestion. 
> git clone <REMOTE> && git checkout <REF>

But I didn't understand where do you want to git clone and git checkout if not from the dockerfile?
Would that be in the opening script (`hack/rp-dev/automate_rp_deploy.sh`) that the docerfile executes?

Reply:
I think that we can just assume the local filesystem has that in place. Either a developer runs it before the make target is executed, or a pipeline does a checkout of a ref/etc.

Reply:
Good suggestions @SudoBrendan! It improved the script's readability and usage.

Reply:
`.dockerignore` doesn't work properly ATM. I will try to tackle that in a follow-up

Reply:
Locally testing with `docker build` and `docker run` it ignored the `.dockerignore`, but when switching to `podman build` and `podman run`, then it worked